spec/unit/settings_spec.rb

```diff
@@ -1478,7 +1478,8 @@ def assert_accessing_setting_is_deprecated(settings, setting)
         expect(catalog.resource(:group, "sgroup")).to be_nil
       end
 
-      it "should not add users or groups to the catalog if :mkusers is a valid setting but is disabled" do
+      it "should not add users or groups to the catalog if :mkusers is a valid setting but is disabled",
+           unless: Puppet::Util::Platform.jruby? do
         @settings[:mkusers] = false
```

Comment:
Since we apply settings catalogs in JRuby I don't think we want to skip this. The problem is that if `mkusers` is false, then puppet tries to see if the `service` user and group exist. To do that it needs to load `user` and `group` providers, which triggers some calls to execing commands.

Reply:
I think this will work:

```diff
diff --git a/spec/unit/settings_spec.rb b/spec/unit/settings_spec.rb
index bb0d383d8c..764e8fec57 100644
--- a/spec/unit/settings_spec.rb
+++ b/spec/unit/settings_spec.rb
@@ -1478,9 +1478,10 @@ describe Puppet::Settings do
         expect(catalog.resource(:group, "sgroup")).to be_nil
       end
 
-      it "should not add users or groups to the catalog if :mkusers is a valid setting but is disabled",
-           unless: Puppet::Util::Platform.jruby? do
+      it "should not add users or groups to the catalog if :mkusers is a valid setting but is disabled" do
         @settings[:mkusers] = false
+        @settings.stubs(:service_user_available?).returns(false)
+        @settings.stubs(:service_group_available?).returns(false)
 
         catalog = @settings.to_catalog
         expect(catalog.resource(:user, "suser")).to be_nil
```

Reply:
Ah, good to know. Updated the PR with your fix.